doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,947 @@
+
+    Returns:
+        A string, providing the vendor's product description of the module.
+```
+
+get_type(self):
+```
+    Retrieves the type of the module.
+
+    Returns:
+        A string, the module-type from one of the predefined types:
+        MODULE_TYPE_SWITCH, MODULE_TYPE_DPU
+```
+
+get_oper_status(self):
+```
+    Retrieves the operational status of the module
+    This information is not sufficient for debugging complex DPU failures.
+    So, couple of new CLIs will be introduced.
+
+    Returns:
+        A string, the operational status of the module from one of the
+        predefined status values: MODULE_STATUS_EMPTY, MODULE_STATUS_OFFLINE,
+        MODULE_STATUS_FAULT, MODULE_STATUS_PRESENT or MODULE_STATUS_ONLINE
+```
+
+reboot(self, reboot_type):
+```
+    Request to reboot the module
+
+    Args:
+        reboot_type: A string, the type of reboot requested from one of the
+        predefined reboot types: MODULE_REBOOT_DEFAULT,
+        MODULE_REBOOT_CPU_COMPLEX, or MODULE_REBOOT_FPGA_COMPLEX
+
+    Returns:
+        bool: True if the request has been issued successfully, False if not
+```
+
+set_admin_state(self, up):
+```
+    Request to keep the card/DPU in administratively up/down state.
+    Default state is down.
+```
+
+get_maximum_consumed_power(self):
+```
+    Retrieves the maximum power drawn by this module
+
+    Returns:
+        A float, with value of the maximum consumable power of the
+        module.
+```
+
+get_midplane_ip(self):
+```
+    Retrieves the midplane IP-address of the module
+    When called from the DPU, returns the midplane IP-address of the dpu-card.
+    When called from the Switch returns the midplane IP-address of Switch.
+
+    Returns:
+        A string, the IP-address of the module reachable over the midplane
+```
+
+is_midplane_reachable(self):
+```
+    Retrieves the reachability status of the module from the Supervisor or
+    of the Supervisor from the module via the midplane of the modular chassis
+
+    Returns:
+        A bool value, should return True if module is reachable via midplane
+```
+#### 3.1.5 ModuleBase class new APIs
+
+##### 3.1.5.1 Need for consistent storage and access of DPU reboot cause, state and health
+1.  The smartswitch needs to know the reboot cause for DPUs. Please refer to the CLI section for the various options and their effects when executed on the switch and DPUs. 
+
+    Table shows the frame work for DPU reboot-cause reporting
+
+    | DPU Reboot Cause | HW/SW | End_User_Message_in_DPU_STATE |
+    | --- | --- | --- |
+    | REBOOT_CAUSE_POWER_LOSS | HW | Power failure |
+    | REBOOT_CAUSE_HOST_RESET_DPU | SW | Host lost DPU - Try resetting DPU |
+    | REBOOT_CAUSE_HOST_POWERCYCLE_DPU | SW | Host lost DPU - Power cycled DPU |
+    | REBOOT_CAUSE_SW_THERMAL |	SW | Switch software Powered Down DPU due to DPU temperature failure |
+    | REBOOT_CAUSE_DPU_SELF_REBOOT | SW | DPU Software reboots the DPU |
```

Comment:
SONiC already has a list of predefined reboot causes: https://github.com/sonic-net/sonic-platform-common/blob/56921d80e42971f2579115f26c5194b9e0ecbc05/sonic_platform_base/chassis_base.py#L20
We need to try to reuse this list for the DPU as well. If from your point of view, it misses causes let's extend the list with the additional causes, rather than defining a completely new list just for DPU.

Reply:
You are right that's the idea. I explicitly called out that these will be added to existing list of "Possible reboot causes" under ChassisBase Class.